xwiki-rendering-macros/xwiki-rendering-macro-html/src/main/java/org/xwiki/rendering/internal/macro/html/HTMLMacroXHTMLRendererFactory.java

```diff
@@ -19,33 +19,90 @@
  */
 package org.xwiki.rendering.internal.macro.html;
 
-import javax.inject.Named;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.inject.Inject;
 import javax.inject.Singleton;
 
 import org.xwiki.component.annotation.Component;
+import org.xwiki.component.descriptor.ComponentDescriptor;
 import org.xwiki.rendering.internal.renderer.AbstractPrintRendererFactory;
 import org.xwiki.rendering.syntax.Syntax;
 import org.xwiki.rendering.syntax.SyntaxType;
 
 /**
- * Create XHTML Renderers for the HTML Macro, see {@link HTMLMacroXHTMLRenderer}.
+ * Create Renderers for the HTML Macro.
+ * It computes the renderer to creates based on the given hint. By default it creates a {@link HTMLMacroXHTMLRenderer}.
  *
  * @version $Id$
  * @since 2.0M3
  */
-@Component
-@Named("xhtmlmacro/1.0")
+@Component(hints = {"xhtmlmacro/1.0", "htmlmacro+annotatedxhtml/1.0"})
```

Comment:
This suggest that you only support xhtml target syntaxes but in view mode the target syntax is almost always html.

Reply:
I only added support for annotatedhtml, all other syntaxes fallback on xhtmlmacro. We can add more support later with custom renderer if needed. The objective in the first place was to be able to handle metadata in annotatedhtml.

Reply:
It would be safer to add support for the currently standard html renderer we provide (so just two more) to not end up with a WTF when we finally decide to switch to annotatedhtml/5.0 in the WYSIWYG (HTML 5 is not well supported by CKEditor 4 from what I understood so we have to stay on XHTML for now).

More importantly for the short term the fallback on exception which will happen for every single HTML macro execution in view mode (since it won't find any corresponding renderer because we use html5) is going to make it a lot slower.

Reply:
> More importantly for the short term the fallback on exception which will happen for every single HTML macro execution in view mode (since it won't find any corresponding renderer because we use html5) is going to make it a lot slower.

Hmmm right I didn't suspect the possible impact on perf. OK I'll add it then.